src/contexts/StateContextProvider.tsx

```diff
@@ -458,10 +459,23 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
     }, []);
 
     // Synchronize `logEventNumRef` with `logEventNum`.
+    // FIXME: is this necessary? Can we integrate it with the [numEvents, logEventNum] useEffect?
     useEffect(() => {
         logEventNumRef.current = logEventNum;
     }, [logEventNum]);
 
+    useEffect(() => {
+        timestampRef.current = timestamp;
+
+        if (null !== mainWorkerRef.current && null !== timestamp) {
+            loadPageByCursor(mainWorkerRef.current, {
+                code: CURSOR_CODE.TIMESTAMP,
+                args: {timestamp: timestamp},
+            });
+            updateWindowUrlHashParams({timestamp: null});
+        }
+    }, [timestamp]);
+
```

Comment:
1. Let's split the `timestampRef.current = timestamp;` into a separate hook like the other ones.
2. Then the rest of the logic should be moved after the other "Synchronize..." useEffect blocks.
3. Let's use early returns for the null checks as done in the other hooks.